### PR TITLE
- Integrate with Chainalysis API

### PR DESCRIPTION
**Pull request by MrQKyber**

### CODE REVIEW

File: KyberNetwork/KyberNetwork/Networking/Internal/KyberNetworkService.swift [hunk 1]
```diff
@@ -285,14 +282,14 @@ extension UserInfoService: TargetType {
...
-    case .getUserTradeCap(let accessToken, _):
-      json["Authorization"] = accessToken
     case .sendTxHash(let accessToken, _):
       json["Authorization"] = accessToken
     case .getNotification(let accessToken, _):
       if let token = accessToken { json["Authorization"] = token }
     case .markAsRead(let accessToken, _):
       if let token = accessToken { json["Authorization"] = token }
+    case .getPreScreeningWallet:
+      return nil
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Networking/Internal/KyberNetworkService.swift]: Should return default json with:
`content-type`, `client`, `client-build`

File: KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift [hunk 2]
```diff
@@ -488,41 +486,16 @@ extension KNExchangeTokenCoordinator: KSwapViewControllerDelegate {
...
-                value: "Sorry, we are unable to handle such a big amount. Please reduce the amount to less than %@ and try again.",
-                comment: ""
-              )
-              errorMessage = String(
-                format: text,
-                "\(display) ETH")
-            }
+          let json = try? resp.mapJSON() as? JSONDictionary ?? [:] {
+          if let status = json["eligible"] as? Bool {
+            print("eligible status : \(status)")
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift]: Should print for debug mode only

File: KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift [hunk 3]
```diff
@@ -488,41 +486,16 @@ extension KNExchangeTokenCoordinator: KSwapViewControllerDelegate {
...
+            errorMessage = json["message"] as? String
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift]: Should first the value of `status` from `json["eligible"]`.
If it is true, `errorMessage` should be set to `nil`, just in case server also returns `message` field for eligible address.

File: KyberNetwork/KyberNetwork/Screens/Tabbars/LimitOrder/Coordinator/KNLimitOrderTabCoordinator.swift [hunk 4]
```diff
@@ -803,16 +803,25 @@ extension KNLimitOrderTabCoordinator: KNConvertSuggestionViewControllerDelegate
...
-          self?.navigationController.hideLoading()
-          self?.navigationController.showErrorTopBannerMessage(
+      sendGetPreScreeningWalletRequest { [weak self] (result) in
+        guard let `self` = self else { return }
+        var message: String?
+        if case .success(let resp) = result,
+          let json = try? resp.mapJSON() as? JSONDictionary ?? [:] {
+          if let status = json["eligible"] as? Bool {
+            print("eligible status : \(status)")
+            message = json["message"] as? String
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/LimitOrder/Coordinator/KNLimitOrderTabCoordinator.swift]: Same for above comment, only cast `message` if status is false

File: KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift [hunk 5]
```diff
@@ -488,41 +486,16 @@ extension KNExchangeTokenCoordinator: KSwapViewControllerDelegate {
...
-              )
-              errorMessage = String(
-                format: text,
-                "\(display) ETH")
-            }
+          let json = try? resp.mapJSON() as? JSONDictionary ?? [:] {
+          if let status = json["eligible"] as? Bool {
+            if isDebug { print("eligible status : \(status)") }
+            if status == false { errorMessage = json["message"] as? String }
           }
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift]: what happen if there is no `json["eligible"]`?

Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift]: it will pass then errorMessage is nil. There will no error 
As your mention before, if this api fail user still have permission to transfer/swap

Comment (manhlx3006) [reply KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/SwapVC/Coordinator/KNExchangeTokenCoordinator.swift]: ok

File: KyberNetwork/KyberNetwork/Screens/Transaction/Send/Coordinator/KNSendTokenViewCoordinator.swift [hunk 6]
```diff
@@ -214,7 +217,26 @@ extension KNSendTokenViewCoordinator: KNSearchTokenViewControllerDelegate {
 extension KNSendTokenViewCoordinator: KConfirmSendViewControllerDelegate {
   func kConfirmSendViewController(_ controller: KConfirmSendViewController, run event: KConfirmViewEvent) {
     if case .confirm(let type) = event, case .transfer(let transaction) = type {
-      self.didConfirmTransfer(transaction)
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Transaction/Send/Coordinator/KNSendTokenViewCoordinator.swift]: It should be added when user clicked to `Transfer Now` button instead of after clicking to `Confirm` in Confirm Transfer view
Before opening the confirmation view, you should call prescreening api to check if address is eligible

File: KyberNetwork/KyberNetwork/Screens/Transaction/Send/Controller/KSendTokenViewController.swift [hunk 7]
```diff
@@ -284,6 +284,7 @@ class KSendTokenViewController: KNBaseViewController {
       transaction: self.viewModel.unconfirmTransaction,
       ens: self.viewModel.isUsingEns ? self.viewModel.addressString : nil
     )
+    sendButton.isEnabled = false
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Transaction/Send/Controller/KSendTokenViewController.swift]: Why the send button is not enabled here?

Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Screens/Transaction/Send/Controller/KSendTokenViewController.swift]: to avoid show multiple confirm screen because loading screening api is a async task so if button is not disable, I can multiple time

File: KyberNetwork/KyberNetwork/Screens/Transaction/Send/Coordinator/KNSendTokenViewCoordinator.swift [hunk 8]
```diff
@@ -123,14 +126,46 @@ extension KNSendTokenViewCoordinator: KSendTokenViewControllerDelegate {
     case .searchToken(let selectedToken):
       self.openSearchToken(selectedToken: selectedToken)
     case .send(let transaction, let ens):
-      self.send(transaction: transaction, ens: ens)
+      sendGetPreScreeningWalletRequest { [weak self] (result) in
```
Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Transaction/Send/Coordinator/KNSendTokenViewCoordinator.swift]: Please add loading indicator here so users will see something is in progress

Comment (manhlx3006) [inline KyberNetwork/KyberNetwork/Screens/Transaction/Send/Coordinator/KNSendTokenViewCoordinator.swift]: Should use `navigationController` or `controller` to show/hide loading indicator here, it will be hard to follow if show and hide logics in different files
